### PR TITLE
Adds ability to make POST Requests

### PR DESCRIPTION
**Pull request by rma-bryson**

### CODE REVIEW

File: cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/HttpRequestBuilderImpl.java [hunk 1]
```diff
@@ -57,10 +66,11 @@ public HttpRequestBuilderImpl(ApiConnectionInfo apiConnectionInfo, String endpoi
         }
         this.httpUrl = url;
         this.endpoint = Objects.requireNonNull(endpoint, "Cannot process request against the API root endpoint");
+        method = HttpRequestMethod.GET; //default method
```
Comment (adamkorynta) [inline cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/HttpRequestBuilderImpl.java:69]: Why do we need a default? Doesn't the fluent API ensure that a method is set?

File: cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/HttpRequestBuilderImpl.java [hunk 2]
```diff
@@ -32,21 +32,30 @@
...
 import okhttp3.Request;
+import okhttp3.RequestBody;
 import okhttp3.Response;
 import okhttp3.ResponseBody;
 import usace.metrics.noop.NoOpTimer;
 import usace.metrics.services.Metrics;
 import usace.metrics.services.Timer;
 
-public class HttpRequestBuilderImpl implements HttpRequestBuilder {
+public final class HttpRequestBuilderImpl implements HttpRequestBuilder {
```
Comment (adamkorynta) [inline cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/HttpRequestBuilderImpl.java:50]: You added the final back in? Does Cumulus not need to extend it?

File: cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/HttpRequestBuilderImpl.java [hunk 3]
```diff
@@ -32,21 +32,30 @@
...
 
     private final String endpoint;
     private final HttpUrl httpUrl;
     private final Map<String, String> queryParameters = new HashMap<>();
     private final Map<String, String> queryHeaders = new HashMap<>();
+    private HttpRequestMethod method;
+    private String body;
+    private String mediaType;
```
Comment (adamkorynta) [inline cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/HttpRequestBuilderImpl.java:58]: How varied is this media type? Can this be an enum?

**Comment by adamkorynta** [comment]
Make sure to bump the version to 1.1.0

File: cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/HttpRequestBuilderImpl.java (hunk 2 above)
Comment (rma-bryson) [reply cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/HttpRequestBuilderImpl.java:50]: Hmm, something might've gotten messed up on the rebase I did. Did you make other changes in the commit where you updated that?

File: cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/HttpRequestBuilderImpl.java (hunk 1 above)
Comment (rma-bryson) [reply cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/HttpRequestBuilderImpl.java:69]: Yeah that's correct. Forgot to remove the default set.

File: cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/HttpRequestBuilderImpl.java (hunk 2 above)
Comment (rma-bryson) [reply cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/HttpRequestBuilderImpl.java:50]: I found the file. I'll do a diff on that, but I agree that it shouldn't be final anymore. I'll update it.

File: cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/HttpRequestBuilderImpl.java (hunk 3 above)
Comment (rma-bryson) [reply cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/HttpRequestBuilderImpl.java:58]: Good question. Looks like there's quite alot of formats that are supported in MediaType.parse(String). We are only supporting different json versions (and possibly some xml), currently however.

Comment (adamkorynta) [reply cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/HttpRequestBuilderImpl.java:58]: Ok, String is file with me

**Review by adamkorynta — APPROVED** [review]
Looks good!

**Pull request merged**